File: components/device-mgt/org.wso2.carbon.device.mgt.common/src/main/java/org/wso2/carbon/device/mgt/common/Device.java
/**
 *  Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */
package org.wso2.carbon.device.mgt.common;

import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;
import java.util.List;

@XmlRootElement
public class Device {

    private int id;
    private String type;
    private String description;
    private String name;
    private Long dateOfEnrolment;
    private Long dateOfLastUpdate;
    private String ownership;
    private boolean status;
    private int deviceTypeId;
    private String deviceIdentifier;
    private String owner;
    private List<Feature> features;
    private List<Device.Property> properties;

    @XmlElement
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @XmlElement
    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @XmlElement
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @XmlElement
    public Long getDateOfEnrolment() {
        return dateOfEnrolment;
    }

    public void setDateOfEnrolment(Long dateOfEnrolment) {
        this.dateOfEnrolment = dateOfEnrolment;
    }

    @XmlElement
    public Long getDateOfLastUpdate() {
        return dateOfLastUpdate;
    }

    public void setDateOfLastUpdate(Long dateOfLastUpdate) {
        this.dateOfLastUpdate = dateOfLastUpdate;
    }

    @XmlElement
    public String getOwnership() {
        return ownership;
    }

    public void setOwnership(String ownership) {
        this.ownership = ownership;
    }

    @XmlElement
    public boolean isStatus() {
        return status;
    }

    public void setStatus(boolean status) {
        this.status = status;
    }

    @XmlElement
    public int getDeviceTypeId() {
        return deviceTypeId;
    }

    public void setDeviceTypeId(int deviceTypeId) {
        this.deviceTypeId = deviceTypeId;
    }

    @XmlElement
    public String getDeviceIdentifier() {
        return deviceIdentifier;
    }

    public void setDeviceIdentifier(String deviceIdentifier) {
        this.deviceIdentifier = deviceIdentifier;
    }

    @XmlElement
    public String getOwner() {
        return owner;
    }

    public void setOwner(String owner) {
        this.owner = owner;
    }

    @XmlElement
    public List<Feature> getFeatures() {
        return features;
    }

    public void setFeatures(List<Feature> features) {
        this.features = features;
    }

    @XmlElement
    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    @XmlElement
    public List<Device.Property> getProperties() {
        return properties;
    }

    public void setProperties(List<Device.Property> properties) {
        this.properties = properties;
    }

    public static class Property {

        private String name;
        private String value;

        public String getName() {
            return name;
        }

        public void setName(String name) {
            this.name = name;
        }

        public String getValue() {
            return value;
        }

        public void setValue(String value) {
            this.value = value;
        }
    }

}

File: product/modules/distribution/src/repository/jaggeryapps/cdm/modules/device.js
/*
 * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
 *
 * WSO2 Inc. licenses this file to you under the Apache License,
 * Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

var utility = require("/modules/utility.js");
var DeviceIdentifier = Packages.org.wso2.carbon.device.mgt.common.DeviceIdentifier;
var log = new Log();

var deviceManagementService = utility.getDeviceManagementService();

var listDevices = function () {
    var devices = deviceManagementService.getAllDevices("android");
    var deviceList = [];

    for (i = 0; i < devices.size(); i++) {
        var device = devices.get(i);
        deviceList.push({
            "identifier": device.getDeviceIdentifier(),
            "name": device.getName(),
            "ownership": device.getOwnership(),
            "owner": device.getOwner(),
            "deviceType": device.getType(),
            "vendor": device.getProperties().get("vendor"),
            "model": device.getProperties().get("model"),
            "osVersion": device.getProperties().get("osVersion")
        });
    }
    return deviceList;
}
var getDevice = function(type, deviceId){
    var deviceIdentifier =  new DeviceIdentifier();
    deviceIdentifier.setType(type);
    deviceIdentifier.setId(deviceId);
    var device = deviceManagementService.getDevice(deviceIdentifier);
    return device;
}

var viewDevice = function(type, deviceId){
    var device = this.getDevice(type, deviceId);

    var entries = device.getProperties().entrySet();
    var iterator = entries.iterator();
    var properties = {};
    while(iterator.hasNext()){
        var entry = iterator.next();
        var key = entry.getKey();
        var value = entry.getValue();
        properties[key]= value;
    }
    return {
        "identifier": device.getDeviceIdentifier(),
        "name": device.getName(),
        "ownership": device.getOwnership(),
        "owner": device.getOwner(),
        "deviceType": device.getType(),
        "vendor": device.getProperties().get("vendor"),
        "model": device.getProperties().get("model"),
        "osVersion": device.getProperties().get("osVersion"),
        "properties": properties
    };
}